lib/spack/spack/config.py

```diff
@@ -758,12 +759,19 @@ def __iter__(self):
         """Iterate over scopes in this configuration."""
         yield from self.scopes.values()
 
-    def print_section(self, section: str, blame: bool = False, *, scope=None) -> None:
+    def print_section(
+        self, section: str, yaml: bool = True, blame: bool = False, *, scope: Optional[str] = None
```

Comment:
PR looks good. The only nitpick is that the `yaml` argument doesn't matter when `blame=True` . This is fine cause with provenance it's not really yaml, but it seems more a tristate "yaml|json|blame". Maybe we should just extend the docstring and explain that?

Reply:
as per construction no code path exists where blame=true and yaml=false, so I think the booleans are simple enough, otherwise it's just extra lines of code

Reply:
> so I think the booleans are simple enough, otherwise it's just extra lines of code

I agree, that's why I suggested docstring in the first place.